controllers/offer.js

```diff
@@ -1,7 +1,13 @@
 const offerService = require('~/services/offer')
 
 const getOffers = async (req, res) => {
-  const offers = await offerService.getOffers()
+  const { catId, subId } = req.params
+
+  const filters = {}
+  if (catId) filters.categoryId = catId
+  if (subId) filters.subjectId = subId
```

Comment:
can't you use getMatchOptions util here?

Reply:
done